pkg/spec/spec.go

```diff
@@ -174,10 +174,20 @@ func (config *CreateConfig) createConfigToOCISpec(runtime *libpod.Runtime, userM
 	}
 
 	hostname := config.Hostname
-	if hostname == "" && (config.NetMode.IsHost() || config.UtsMode.IsHost()) {
-		hostname, err = os.Hostname()
-		if err != nil {
-			return nil, errors.Wrap(err, "unable to retrieve hostname")
+	if hostname == "" {
+		if utsCtrID := config.UtsMode.Container(); utsCtrID != "" {
+			utsCtr, err := runtime.GetContainer(utsCtrID)
+			if err != nil {
+				return nil, errors.Wrapf(err, "unable to retrieve hostname from dependency container %s", utsCtrID)
+			}
+			hostname = utsCtr.Hostname()
+		} else if config.NetMode.IsHost() || config.UtsMode.IsHost() {
+			hostname, err = os.Hostname()
+			if err != nil {
+				return nil, errors.Wrap(err, "unable to retrieve hostname")
```

Comment:
Logging the attempted hostname would help with debugging.

Reply:
I would expect hostname to be empty here, because we errored on a call to os.Hostname(). I certainly don't expect it to have anything of value, but I can note that it failed to get the host's hostname